src/squeeze_xen.ml

```diff
@@ -86,6 +86,22 @@ module Domain = struct
 			  Hashtbl.remove cache domid;
 			  None
 
+  let remove_gone_domains_cache xc =
+	  let current_domains = Xenctrl.domain_getinfolist xc 0 in
+	  Mutex.execute m
+	  (fun () ->
+			try
```

Comment:
Is this `try ... with` needed? I don't think the code it guards can throw an exception.